arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -1143,6 +1265,357 @@ IResearchAnalyzerFeature::AnalyzerPool::ptr IResearchAnalyzerFeature::get( // fi
+    auto* engine = arangodb::EngineSelectorFeature::ENGINE;
+    auto itr = _lastLoad.find(databaseKey); // find last update timestamp
+
+    if (!engine || engine->inRecovery()) {
+      // always load if inRecovery since collection contents might have changed
+    } else if (arangodb::ServerState::instance()->isSingleServer()) { // single server
+      if(itr != _lastLoad.end()) {
+        return arangodb::Result(); // do not reload on single-server
+      }
+    } else if (itr != _lastLoad.end() // had a previous load
+               && itr->second + RELOAD_INTERVAL > currentTimestamp // timeout not reached
+              ) {
+      return arangodb::Result(); // reload interval not reached
+    }
+
+    auto* vocbase = dbFeature->lookupDatabase(database);
+    static auto cleanupAnalyzers = []( // remove invalid analyzers
+      IResearchAnalyzerFeature& feature, // analyzer feature
+      decltype(_lastLoad)::iterator& lastLoadItr, // iterator
+      irs::string_ref const& database // database
+    )->void {
+      if (lastLoadItr == feature._lastLoad.end()) {
+        return; // nothing to do (if not in '_lastLoad' then not in '_analyzers')
+      }
+
+      // remove invalid database and analyzers
+      feature._lastLoad.erase(lastLoadItr);
+
+      // remove no longer valid analyzers (force remove)
+      for (auto itr = feature._analyzers.begin(),
+           end = feature._analyzers.end();
+           itr != end;
+          ) {
+        auto split = splitAnalyzerName(itr->first);
+
+        if (split.first == database) {
+          itr = feature._analyzers.erase(itr);
+        } else {
+          ++itr;
+        }
+      }
+    };
+
+    if (!vocbase) {
+      cleanupAnalyzers(*this, itr, database); // cleanup any analyzers for 'database'
+
+      return arangodb::Result( // result
+        TRI_ERROR_ARANGO_DATABASE_NOT_FOUND, // code
+        std::string("failed to find database '") + std::string(database) + " while loading analyzers"
+      );
+    } else {
+      std::shared_ptr<arangodb::LogicalCollection> collection;
+      auto collectionCallback = [&collection]( // store collection
+        std::shared_ptr<arangodb::LogicalCollection> const& col // args
+      )->void {
+        collection = col;
+      };
+
+      arangodb::methods::Collections::lookup( // find collection
+        *vocbase, ANALYZER_COLLECTION_NAME, collectionCallback // args
+      );
+
+      if (!collection) {
+        cleanupAnalyzers(*this, itr, database); // cleanup any analyzers for 'database'
+        _lastLoad[databaseKey] = currentTimestamp; // update timestamp
+
+        return arangodb::Result(); // no collection means nothing to load
+      }
+    }
+
+    Analyzers analyzers;
+    auto visitor = [this, &analyzers, &vocbase]( // visitor
+      arangodb::velocypack::Slice const& slice // analyzer definition
+    )->arangodb::Result {
+      if (!slice.isObject()) {
+        LOG_TOPIC(,ERR, arangodb::iresearch::TOPIC)
```

Comment:
log ID missing here.

Reply:
it's empty

Reply:
see above. This is not how it is supposed to be used.